webapp/channels/src/components/drafts/scheduled_post_list/non_virtualized_scheduled_post_list.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {useRef} from 'react';
+
+import type {ScheduledPost} from '@mattermost/types/schedule_post';
+import type {UserProfile, UserStatus} from '@mattermost/types/users';
+
+import DraftRow from 'components/drafts/draft_row';
+
+import {useQuery} from 'utils/http_utils';
+
+type Props = {
+    scheduledPosts: ScheduledPost[];
+    currentUser: UserProfile;
+    userDisplayName: string;
+    userStatus: UserStatus['status'];
+}
+
+export default function NonVirtualizedScheduledPostList(props: Props) {
+    const query = useQuery();
+    const targetId = query.get('target_id');
+    const targetScheduledPostId = useRef<string>();
+
+    return (
+        <>
+            {
+                props.scheduledPosts.map((scheduledPost, index) => {
+                    // find the first scheduled posst with the target and no error
+                    const isInTargetChannelOrThread = scheduledPost.channel_id === targetId || scheduledPost.root_id === targetId;
+                    const hasError = Boolean(scheduledPost.error_code);
+                    const scrollIntoView = !targetScheduledPostId.current && (isInTargetChannelOrThread && !hasError);
+                    if (scrollIntoView) {
+                        // if found, save the scheduled post's ID
+                        targetScheduledPostId.current = scheduledPost.id;
```

Comment:
I see that this comes from the previous version of the code, so there's nothing to change here for now, but this looks strange to me, especially because it's setting a ref inside the body of the component instead of a `useEffect` or in a callback.

This can be left to the followup, but I'd expect for this to be done using a `useEffect` on mount instead of passing a prop to control this.

Reply:
Yeah i had changed this before, but since @harshilsharma63 pointed out a a major bug with the scrolling to a post i had since reverted to the previous implementation for this one